jailer/src/lib.rs

```diff
@@ -97,6 +96,122 @@ pub enum Error {
     Write(PathBuf, io::Error),
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::Error::*;
+
+        match *self {
+            Canonicalize(ref path, ref io_err) => {
+                write!(f, "Failed to canonicalize path {:?}: {:?}", path, io_err)
```

Comment:
```suggestion
                write!(f, "Failed to canonicalize path {:?}: {}", path, io_err)
```

Reply:
The inner error is an `io::Error`: `Canonicalize(PathBuf, io::Error)` - which doesn't implement `Display`, so formatting with `{}` wont't work.